compiler.pri

```diff
@@ -227,39 +227,61 @@ unix {
 		QMAKE_OBJECTIVE_CXXFLAGS *= -O3 -march=native -ffast-math -ftree-vectorize -fprofile-use
 	}
 
-	CONFIG(c++11) {
-		# Qt 5 will pass the expected compiler flags
-		# needed for C++11 mode when CONFIG includes c++11.
-		# But Qt 4 won't, so we add it manually.
-		lessThan(QT_MAJOR_VERSION, 5) {
-			QMAKE_CXXFLAGS *= -std=c++11
+	# First, auto-populate CONFIG with the latest C++ standard
+	# that Qt has detected support for.
+	!CONFIG(c++03):!CONFIG(c++11):!CONFIG(c++14):!CONFIG(c++1z) {
+		contains(QT_CONFIG, c++11): CONFIG *= c++11
+		contains(QT_CONFIG, c++14): CONFIG *= c++14
+		contains(QT_CONFIG, c++1z): CONFIG *= c++1z
+	}
```

Comment:
If QT is compiled with support for more than one of these, then more than one `CONFIG` option is set.

This causes compile errors in the `lessThan(QT_MAJOR_VERSION, 5) {` block.

If qmake has a break statement, I would suggest putting breaks in between each of these lines.  If not, you are going to have to separate this out into three blocks.